aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -690,47 +698,43 @@ AT_ERROR("inverse: MAGMA library not found in "
 }
 
 template <typename scalar_t>
-static void apply_single_inverse(Tensor& self, int64_t& info) {
-#ifndef USE_MAGMA
-AT_ERROR("inverse: MAGMA library not found in "
-    "compilation. Please rebuild with MAGMA.");
-#else
+static void apply_single_inverse(const Tensor& self, int64_t& info, Tensor& ret) {
   auto self_data = self.data_ptr<scalar_t>();
-  magma_int_t n = magma_int_cast(self.size(-2), "self.size(-2)");
-  magma_int_t lwork = n * magmaGetriOptimalBlocksize<scalar_t>(n);
-  magma_int_t info_tmp = 0;
+  int n = magma_int_cast(self.size(-2), "self.size(-2)");
+  int info_tmp = 0;
 
-  Tensor ipiv = at::empty({n}, at::kInt);
-  Tensor dwork = at::empty({lwork}, self.options());
-  magmaLu<scalar_t>(n, n, self_data, n, ipiv.data_ptr<magma_int_t>(), &info_tmp);
+  Tensor ipiv = at::empty({n}, self.options().dtype(at::kInt));
+  cu_xgetrf<scalar_t>(n, n, self_data, n, ipiv.data_ptr<int>(), &info_tmp);
   if (info_tmp != 0) {
     info = info_tmp;
     return;
   }
-  magmaGetri<scalar_t>(
-    n, self_data, n, ipiv.data_ptr<magma_int_t>(), dwork.data_ptr<scalar_t>(), lwork, &info_tmp);
+  ret = at::eye(n, self.options());
+  cu_xgetri<scalar_t>(n, self_data, n, ipiv.data_ptr<int>(), &info_tmp, ret);
   info = info_tmp;
-#endif
 }
 
 Tensor _inverse_helper_cuda(const Tensor& self) {
-  auto self_inv_working_copy = cloneBatchedColumnMajor(self);
   if (self.dim() > 2) {
+    auto self_inv_working_copy = cloneBatchedColumnMajor(self);
     std::vector<int64_t> infos(batchCount(self), 0);
     auto self_working_copy = cloneBatchedColumnMajor(self);
     AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "inverse_cuda", [&]{
       apply_batched_inverse<scalar_t>(
         self_working_copy, self_inv_working_copy, infos);
     });
     batchCheckErrors(infos, "inverse_cuda");
+
+    return self_inv_working_copy;
   } else {
+    Tensor ret;
     int64_t info = 0;
     AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "inverse_cuda", [&]{
-      apply_single_inverse<scalar_t>(self_inv_working_copy, info);
+      apply_single_inverse<scalar_t>(self.clone(at::MemoryFormat::Contiguous).transpose_(-2, -1), info, ret);
```

Comment:
Can't you simply pass `self_inv_working_copy`. It's still going to be column major.

Reply:
Yes, this is just a temporary workaround to test functionality. Later I'll also update batched inverse code, then I will use self_inv_working_copy.